app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp

```diff
@@ -1098,6 +1100,17 @@ DeviceDelegateOculusVR::SetRenderMode(const device::RenderMode aMode) {
   vrb::RenderContextPtr render = m.context.lock();
   for (int i = 0; i < VRAPI_EYE_COUNT; ++i) {
     m.eyeSwapChains[i]->Init(render, m.renderMode, m.renderWidth, m.renderHeight);
+    // auto surfaceOut = vrapi_GetTextureSwapChainAndroidSurface(m.eyeSwapChains[i]->ovrSwapChain);
+    // surfaceOut is null from vrapi_GetTextureSwapChainAndroidSurface(m.eyeSwapChains[i]->ovrSwapChain);.
+  }
+
+  for (int i = 0; i < DeviceDelegateOculusVR::State::EYE_SURFACE_COUNT; ++i) {
+    if (!m.eyeSurfaceSwapChain[i]) {
+      m.eyeSurfaceSwapChain[i] = vrapi_CreateAndroidSurfaceSwapChain(m.renderWidth, m.renderHeight);
+      auto surfaceOut = vrapi_GetTextureSwapChainAndroidSurface(m.eyeSurfaceSwapChain[i]);
+      surfaceOut = m.java.Env->NewGlobalRef(surfaceOut);
```

Comment:
swapChains and surface needs to be deallocated each time the app goes to background an resumes, check `DeviceDelegateOculusVR::EnterVR` and `DeviceDelegateOculusVR::LeaveVR`

Reply:
It looks like I just need to deallocate the `m.eyeSurfaceSwapChain` at `LeaveVR()`, and It will be recreated at ` SetRenderMode()`.